src/main/scala/com/lightbend/kafka/scala/streams/typesafe/unsafe.scala

```diff
@@ -0,0 +1,86 @@
+  *
+  * @author Santiago Saavedra (ssaavedra@openshine.com)
+  */
+object unsafe {
+
+  trait ConverterToTypeSafer[-Src[K, V], +Dst[K, V]] {
+    def safe[K, V](src: Src[K, V]): Dst[K, V]
+  }
+
+  /** This class provides `.unsafely` instances for
+    * [[com.lightbend.kafka.scala.streams.typesafe.TSKType]] values that may be
+    * needed to provide values that cannot be assumed to be type-safe. You
+    * can also use the `.unsafelyNoWrap` construct which does not wrap the
+    * returned type (and thus you will have an abstraction leak, but you
+    * needn't provide the type conversion).
+    *
+    * @param inner the TSKType that is being unsafely handled
+    * @tparam KType the underlying implementation type (A KStream/KTable/etc)
+    * @tparam K     the type of the key for this TSKType
+    * @tparam V     the type of values for this TSKType
+    */
+  implicit class UnsafelyWrapper[KType[_, _], K, V]
+  (private val inner: TSKType[KType, K, V])
+  extends AnyVal {
+
+    /** Allows the user to perform potentially-unsafe operations with the
+      * TSKType, which may be usesful for accessing properties not yet
+      * implemented on this library.
+      *
+      * Take note that in order not to leak the type-unsound abstraction you
+      * must provide a [[ConverterToTypeSafer]] which should wrap the
+      * value of `transformer` into its returned type. Your provided function
+      * should operate only on the underlying types.
+      *
+      * Take note that it is assumed as of now that your operation will
+      * return a (*, *)-kind type, with "holes" for a resulting key and value.
+      * If you can't operate under such constraint, you will need to rely on
+      * [[unsafelyNoWrap()]] for the moment.
+      *
+      * @param transformer the function that transforms the inner type
+      * @param wrap        the evidence used to wrap the transformer output
+      *                    back to a typesafe value (needn't be a TSKType, so
+      *                    you can implement your own conversions)
+      * @tparam K1        the resulting key type
+      * @tparam V1        the resulting value type
+      * @tparam UnsafeDst the type returned by the transformer, which is a
+      *                   raw type in the underlying implementation
+      * @tparam SafeDst   the type that should be returned after performing the
+      *                   unsafe operation and which should wrap the type
+      *                   returned by the transformer in a more
+      *                   type-constrained way
+      * @return a SafeDst value that has been transformed by the transformer
+      *         on its underlying values
+      */
+    def unsafely[K1, V1, UnsafeDst[_, _], SafeDst[_, _]]
+    (transformer: KType[K, V] => UnsafeDst[K1, V1])
+    (implicit wrap: ConverterToTypeSafer[UnsafeDst, SafeDst])
+    : SafeDst[K1, V1] = wrap.safe(transformer(inner.unsafe))
```

Comment:
Will this `unsafely` work for an underlying function like `KStream#to` which returns a `Unit` ?

Reply:
No. The type signature is a bit convoluted, but from the `transformer: KType[K, V] => UnsafeDst[K1, V1]` you see that this is different from the return type of `unsafely`, namely `SafeDst[K1, V1]`.

In this case, `unsafely` expects to have an instance of `ConverterToTypeSafer[UnsafeDst, SafeDst]` which means that there must be a conversion from UnsafeDst to SafeDst available.

I tried to convey this in the Scaladoc but I may have failed, if you have a better wording for this it's much welcome :)

Reply:
Does this mean we cannot use the default serdes for methods like `KStream#to` which returns a `Unit` ?

Reply:
For that you will have to use `unsafelyNoWrap`, since I thought of `unsafely` as a "chaining" operator, which would still return an object to operate with. I like to keep `unsafely` because it already provides the wrap back to a TSKType object.

Reply:
May be I was missing something and could not get through. Will u please change one of the `KStream#to` calls to use this feature of using default config ?

Reply:
I don't understand what you mean here. You want to expose in this API a `TSKStream.toWithDefautSerdes(topic: String)` ?

Reply:
You can already do

`stream.unsafelyNoWrap(_.to(topic))`